### PR TITLE
plan: get correct columns

### PR DESCRIPTION
**Pull request by zimulala**
#5818

### CODE REVIEW

File: model/model.go [hunk 1]
```diff
@@ -181,6 +181,22 @@ func (t *TableInfo) GetPkColInfo() *ColumnInfo {
 	return nil
 }
 
+// Cols returns the columns of the table in public state.
+func (t *TableInfo) Cols() []*ColumnInfo {
```
Comment (shenli) [inline model/model.go]: This will be called many times. Could you pre-calculate this?

Comment (winkyao) [reply model/model.go]: you can save the publicColumns in TableInfo.

Comment (coocood) [reply model/model.go]: It's only called in `ToPB`.
Later we can change the order of columns in DDL, then we don't need to allocate memories.

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**